src/figuresClasses.ts

```diff
@@ -1,19 +1,62 @@
-export interface Figure {
+type Shape = 'triangle' |'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
-
+  shape: Shape = 'triangle';
+
+  constructor(public color: Color,
+    public a: number,
+    public b: number,
+    public c: number) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle sides');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return +(Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)))
+      .toFixed(2);
+  }
 }
 
 export class Circle implements Figure {
+  shape: Shape = 'circle';
 
+  constructor(public color: Color, public radius: number) {
+    if (radius <= 0) {
+      throw new Error('Invalid circle radius');
```

Comment:
In general it is better to give a more detailed explanation of why the error occurred in order to understand how to fix it

for example: 'Invalid circle radius. Radius must be more than 0'

Check all your error messages